ddtrace/contrib/flask/patch.py

```diff
@@ -279,6 +284,48 @@ def traced_start_response(status_code, headers):
     return traced_start_response
 
 
+@with_instance_pin
+def traced_stream_with_context(pin, wrapped, instance, args, kwargs):
+    """
+    Wrapper for flask.stream_with_context
+
+    This function wraps streamed responses.
+    Note - Responses are stream after the flask wsgi app is called. Therefore the span produced by this
+    wrapper will be a sibling span of flask.request.
+    """
+    # stream_with_context(...) copies the current request context before returning an iterable.
+    # To avoid raising a RuntimeError in the wsgi server we must wrap the iterable returned by
+    # this function: https://github.com/pallets/flask/blob/2.1.2/src/flask/helpers.py#L113
+    iterable = wrapped(*args, **kwargs)
+
+    resource = u" ".join((flask.request.method, flask.request.path))
+
+    def _wraped_generator():
+        # TO FIX: GeneratorExit is called when the generator returned stream_with_context(...) is garbage collected.
+        # Calling span.finish() at the point might work in CPython but this could cause issues in other python
+        # implementation (ex. PyPy) or when a reference to the generator persists well after the final iteration.
+        span = pin.tracer.trace(
+            "flask.streamed_response",
+            service=trace_utils.int_service(pin, config.flask),
+            resource=resource,
+            span_type=SpanTypes.WEB,
+        )
+        try:
+            while True:
+                yield iterable.__next__()
+        except StopIteration:
+            # StopIteration inherits from Exception while GeneratorExit inherits from BaseException.
+            # Do not set span.error if StopIteration is raised. This is expected.
+            pass
```

Comment:
oh, no... it isn't.... but.... still not sure if that is what we want?

if this try/except was inside the `while True`, then it would cause an infinite loop, but then the way this is our `Stopiteration` will get raised instead of `iterable`s `StopIteration` which might be fine... ?